coverage/PENDING_REVIEW/intel/0451-COVERAGE-nir-loop-analyze-834-937-995/cov-texel-double-negation.amber

```diff
@@ -0,0 +1,286 @@
+#!amber
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific NIR code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex texgen_vert PASSTHROUGH
+
+SHADER fragment texgen_frag GLSL
+#version 430
+precision highp float;
+
+layout(location = 0) out vec4 _GLF_color;
+
+void main()
+{
+ _GLF_color = vec4(
+ gl_FragCoord.x * (1.0 / 256.0),
+ (int(gl_FragCoord.x) ^ int(gl_FragCoord.y)) * (1.0 / 256.0),
+ gl_FragCoord.y * (1.0 / 256.0),
+ 1.0);
```

Comment:
I don't understand why `gl_FragCoord.x * (1.0 / 256.0)` is guaranteed to come out as 1.0 for all fragments and `gl_FragCoord.y * (1.0 / 256.0)` guaranteed to come out as 0.0 for all fragments. Surely if there are two fragments whose x and y coordinates are swapped, the result of one of these expressions for the first fragment would be the same as the result of the other expression for the second fragment.

I also find the ^ logic hard to follow and would appreciate some comments to explain.

Reply:
You're looking at the texture generation code provided by gfauto. The manually reduced shader is below that.

Reply:
Haha, thanks for clarifying!